plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/file/FileHiveMetastore.java

```diff
@@ -203,6 +203,12 @@ public synchronized void createDatabase(HiveIdentity identity, Database database
 
         Path databaseMetadataDirectory = getDatabaseMetadataDirectory(database.getDatabaseName());
         writeSchemaFile(DATABASE, databaseMetadataDirectory, databaseCodec, new DatabaseMetadata(currentVersion, database), false);
+        try {
+            metadataFileSystem.mkdirs(databaseMetadataDirectory);
```

Comment:
Why is it needed in `Don't put database schema files in the database directories` commit?

Reply:
It's ensuring that the database directory exists when the database is created. It's not strictly necessary, but I think it makes sense.